config/vendor-functions/show-vendor-menu.sh

```diff
@@ -12,8 +12,21 @@ if [[ $(tty) = /dev/tty1 ]] && [[ -f "/home/REKEY_VIA_TPM" ]]; then
   sudo "${VX_FUNCTIONS_ROOT}/rekey-via-tpm.sh"
 fi
 
-if [[ $(tty) = /dev/tty1 ]] && [[ -f "${VX_CONFIG_ROOT}/RUN_BASIC_CONFIGURATION_ON_NEXT_BOOT" ]]; then
+# Note: EXPAND_VAR will be created as part of a vx-iso install
+# This prevents the var expansion from running in VMs while other
+# config/setup may be taking place. We only want to expand var
+# on systems installed to physical hardware.
+if [[ $(tty) = /dev/tty1 ]] && [[ -f "${VX_CONFIG_ROOT}/EXPAND_VAR" ]]; then
```

Comment:
Noting that we're going to add an `|| [[ -f "${VX_CONFIG_ROOT}/RUN_BASIC_CONFIGURATION_ON_NEXT_BOOT" ]]` to make sure that, when using a new image with an old vx-iso, we still properly expand the var partition

Reply:
There was no good way to solve this via the vendor menu / autologin approach, so I instead set the `EXPAND_VAR` flag file via setup-machine.sh. New vx-iso releases also set it during imaging, so we can eventually remove it, but this will make it simple for old vx-iso releases to still work with the new approach.